src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.Windows.cs

```diff
@@ -126,6 +127,24 @@ private bool SetPriorityLive(ThreadPriority priority)
             return Interop.Kernel32.SetThreadPriority(_osHandle, (int)MapToOSPriority(priority));
         }
 
+        [UnmanagedCallersOnly]
+        private static void OnThreadExit()
```

Comment:
@jkotas since we implemented thread shutdown callback are `if ((state & (int)(ThreadState.Unstarted | ThreadState.Stopped | ThreadState.Aborted)) == 0)` and `JoinInternal(0)` still needed in `GetThreadState()` ?

Reply:
Right, it should not be needed.